src/packagedcode/maven.py

```diff
@@ -145,6 +146,33 @@ def api_data_url(self, baseurl=default_api_baseurl):
             filename=filename,
             baseurl=baseurl)
 
+    def compute_normalized_license(self):
+        """
+        Get the license expression by combing the expression with AND of all licenses, and in each license, combining the expression with AND from detected value from name, url and comments.
+        """
+        if self.declared_license:
+            detected_licenses = []
+            for license_declaration in self.declared_license:
+                detected_from_name = license_declaration.get('name')
+                detected_from_url = license_declaration.get('url')
+                detected_from_comments = license_declaration.get('comments')
+                # 1. try detection on the value of name if not empty and keep this
+                # 2. try detection on the value of url  if not empty and keep this
+                # 3. try detection on the value of comment  if not empty and keep this
+                # 4. if the three detection are for the same license, this becomes the kept license for that one licenses item
+                # 5. if not, the name should have precedence and any unknowns
+                # in url and comment should be ignored.
+                detected_items = [models.compute_normalized_license(item) for item in (detected_from_name, detected_from_url, detected_from_comments) if item and item != 'unknown']
+                if not detected_items:
+                    # all detected value from name, url and comments are
+                    # unknown, we should return unknown
+                    license_detected = 'unknown'
+                else:
+                    # use and expression to combine detected values from name,
+                    # url and comment
+                    license_detected = models.compute_normalized_license(' AND '.join(detected_items))
```

Comment:
this is not how you combine tow license expressions. You need to create a Licensing() and then parse the two expressions and create a new And(exp1, exp2) OR ... `' AND '.join(detected_items)` as you did here and parse it with a Licensing(). You cannot invoke `compute_normalized_license` on them